diff --git i/cmake/cmake_findExternalLibs.cmake w/cmake/cmake_findExternalLibs.cmake
index 610170a88c..19e12aab6c 100644
--- i/cmake/cmake_findExternalLibs.cmake
+++ w/cmake/cmake_findExternalLibs.cmake
@@ -367,6 +367,8 @@ if (WITH_OPENTIMS)
   if(Opentims_FOUND)
     message(STATUS "opentims: using system installation (${Opentims_LIBRARIES})")
 
+    find_package(zstd REQUIRED)
+
     # If the installed library was built with OPENTIMS_LINK_SQLITE_STATICALLY,
     # that compile definition is propagated via the imported target's
     # INTERFACE_COMPILE_DEFINITIONS. OpenMS must then supply sqlite3 headers
diff --git i/src/openms/CMakeLists.txt w/src/openms/CMakeLists.txt
index c71713a091..b35b42da6e 100644
--- i/src/openms/CMakeLists.txt
+++ w/src/openms/CMakeLists.txt
@@ -97,6 +97,12 @@ if (OPENMS_ARROW_DATASET_TARGET)
 endif()
 
 if (WITH_OPENTIMS)
+  if(TARGET zstd::libzstd_shared)
+    list(APPEND OPENMS_DEP_PRIVATE_LIBRARIES zstd::libzstd_shared)
+  elseif(TARGET zstd::libzstd_static)
+    list(APPEND OPENMS_DEP_PRIVATE_LIBRARIES zstd::libzstd_static)
+  endif()
+
   list(APPEND OPENMS_DEP_PRIVATE_LIBRARIES opentims::opentims_cpp)
 endif()
 
