diff --git i/CMakeLists.txt w/CMakeLists.txt
index 6c66799..adccb61 100644
--- i/CMakeLists.txt
+++ w/CMakeLists.txt
@@ -70,7 +70,11 @@ endmacro()
 #------------------------------------------------------------------------------
 if(OPENTIMS_BUILD_CPP_LIB)
     add_library(opentims_cpp STATIC ${OPENTIMS_CPP_SOURCES})
-    target_include_directories(opentims_cpp PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}/src")
+    target_include_directories(opentims_cpp
+      PUBLIC
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/src>  # or include/
+      $<INSTALL_INTERFACE:include>
+    )
     opentims_compile_options(opentims_cpp)
     set_target_properties(opentims_cpp PROPERTIES POSITION_INDEPENDENT_CODE ON)
 
diff --git i/src/opentims++/sqlite_helper.h w/src/opentims++/sqlite_helper.h
index 9560a45..d79206b 100644
--- i/src/opentims++/sqlite_helper.h
+++ w/src/opentims++/sqlite_helper.h
@@ -8,10 +8,13 @@
 #include <sqlite3.h>
 #include <string>
 #include <stdexcept>
+#include <optional>
+#include "so_manager.h"
 
 class ot_sqlite
 {
 public:
+    static std::optional<LoadedLibraryHandle> sqlite_so_handle;
     static int sqlite3_open_v2(const char* s, sqlite3** ptr, int flags, const char*) { return ::sqlite3_open_v2(s, ptr, flags, NULL); }
     static int sqlite3_close(sqlite3* db) { return ::sqlite3_close(db); }
     static int sqlite3_exec(sqlite3* db, const char* query, int (*callback)(void*,int,char**,char**), void* arg, char **err) { return ::sqlite3_exec(db, query, callback, arg, err); }
