diff --git i/src/pyOpenMS/CMakeLists.txt w/src/pyOpenMS/CMakeLists.txt
index 550e1b8778..11a673f12f 100644
--- i/src/pyOpenMS/CMakeLists.txt
+++ w/src/pyOpenMS/CMakeLists.txt
@@ -405,7 +405,7 @@ else()
     if(APPLE)
       add_custom_target(pyopenms_fix_deps ALL
         COMMAND ${PROJECT_SOURCE_DIR}/mac_fix_dependencies.rb -l "${PYOPENMS_BUILD_DIR}/pyopenms" -e "@rpath/" -r "@loader_path/" -f -v
-        COMMAND codesign --force --sign - ${PYOPENMS_BUILD_DIR}/pyopenms/Qt* || true
+        # COMMAND codesign --force --sign - ${PYOPENMS_BUILD_DIR}/pyopenms/Qt* || true
         COMMAND codesign --force --sign - ${PYOPENMS_BUILD_DIR}/pyopenms/*.dylib || true
         COMMENT "Fixing deps and codesigning pyopenms binaries in build dir"
       )
